test/alternator/test_cql_rbac.py

```diff
@@ -724,3 +724,20 @@ def test_rbac_tagresource(dynamodb, cql):
                 with temporary_grant(cql, 'ALTER', cql_table_name(table), role):
                     d.meta.client.tag_resource(ResourceArn=arn, Tags=tags)
                     d.meta.client.untag_resource(ResourceArn=arn, TagKeys=['hello'])
+
+# Test that UpdateTimeToLive requires the ALTER permissions, similar to
+# UpdateTable.
+def test_rbac_updatetimetolive(dynamodb, cql, test_table):
+    with new_role(cql) as (role, key):
+        with new_dynamodb(dynamodb, role, key) as d:
+            # Without ALTER permissions, UpdateTimeToLive will fail with
+            # AccessDeniedException. With the ALTER permissions, it will
+            # still fail, but with a ValidationException (because we try
+            # to disable TTL that wasn't enabled).
+            with pytest.raises(ClientError, match='AccessDeniedException'):
+                d.meta.client.update_time_to_live(TableName=test_table.name,
+                    TimeToLiveSpecification={'AttributeName': 'dog', 'Enabled': False})
+            with temporary_grant(cql, 'ALTER', cql_table_name(test_table), role):
+                with pytest.raises(ClientError, match='ValidationException'):
+                    d.meta.client.update_time_to_live(TableName=test_table.name,
+                        TimeToLiveSpecification={'AttributeName': 'dog', 'Enabled': False})
```

Comment:
I'm a bit concerned that we're not testing a successful scenario (meaning TTL has been changed when we have sufficient permissions) but only rely on a priority of validating errors, which is pretty white-box-testing-based.

Reply:
I think explained this in an earlier test - in ALL tests in this file I could check the correctness of the operation after the validation passed, but I felt it was excessive. I know that the permission checks either throw an error, or they don't. How could they possibly NOT throw an error and yet break the normal behavior of the function?

Reply:
I thought about a bug where you reject TTL modifications if a role has or lacks permissions. We test the case where a role lacks permissions and cannot perform an operation, but we're not testing if a role with permissions can still modify TTL.

Reply:
The test (test_rbac_updatetimetolive) does test UpdateTimeToLive both in the no permissions and with-permissions case. But the confusing thing is that even in the with-permissions case we get an error (ValidationExeption) because the code tries to **disable** TTL when it wasn't already disabled.

The not-so-pretty thing about this test is that it relies on the implementation to check the permissions first and only then the validationexception. Which in today's implementation is correct, so this test is good, but if in the future we change the order of these tests, this test will stop testing anything.

So seeing you didn't like it, I'll change it now and make the test really **enable** TTL, which should really work. It will make the test uglier (we need to disable tablets to disable ttl) but it will make it more future-proof.

Reply:
Done. Now the test has Enabled=TRUE and the UpdateTimeToLive is expected to succeed, not fail on some other validation error, so it's better.